tool/speed.cc

```diff
@@ -1022,6 +1022,33 @@ static bool SpeedECDHCurve(const std::string &name, int nid,
   return true;
 }
 
+static bool SpeedECGenCurve(const std::string &name, int nid,
+                            const std::string &selected) {
+  if (!selected.empty() && name.find(selected) == std::string::npos) {
+    return true;
+  }
+
+  BM_NAMESPACE::UniquePtr<EC_KEY> key(EC_KEY_new_by_curve_name(nid));
+  if(!key) {
+    return false;
+  }
+
+  TimeResults results;
+  if (!TimeFunction(&results, [&key]() -> bool {
+// We run extra checks on the key in FIPS mode
+#if defined(BORINGSSL_FIPS)
+        return EC_KEY_generate_key_fips(key.get()) == 1;
+#else
+        return EC_KEY_generate_key(key.get()) == 1;
+#endif
```

Comment:
both functions are always available if I'm not mistaken (regardless of the FIPS flag), so why not measure and report performance for both?

Reply:
We could do that, I just figured we'd want fips for fips, and non-fips for non-fips. If you think this is valuable I'll make a quick change for it